x-pack/legacy/plugins/siem/public/components/ml_popover/api.tsx

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import chrome from 'ui/chrome';
+import { Group, MlSetupArgs } from './types';
+
+/**
+ * Fetches ML Groups Data
+ *
+ * @param headers
+ */
+export const groupsData = async (headers: Record<string, string | undefined>): Promise<Group[]> => {
+  const response = await fetch('/api/ml/jobs/groups', {
+    method: 'GET',
+    credentials: 'same-origin',
+    headers: {
+      'kbn-system-api': 'true',
+      'content-Type': 'application/json',
+      'kbn-xsrf': chrome.getXsrfToken(),
+      ...headers,
+    },
+  });
+  return await response.json();
```

Comment:
Optional:
For now, you might want to wrap this in a try/catch and put a TODO: about the error toaster pop up. This endpoint could end not existing or there could be a race condition on a server restart where the end point isn't available.

Then we can write a ticket to integrate with the error toaster. I did a try/catch for mine fwiw.

Reply:
Followed suit with how you handled errors in https://github.com/elastic/kibana/pull/39483 -- thanks!